Ktisis/Interface/Components/BoneTree.cs

```diff
@@ -32,30 +49,49 @@ public class BoneTree {
 				}
 			}
 		}
+
+		private static bool BoneDoesMatch(Bone bone) => 
+			SearchText != "" && (new [] { bone.UniqueName, bone.LocaleName }).Any(
+				name => name.Contains(SearchText, StringComparison.CurrentCultureIgnoreCase)
+			);
+
 		private static void DrawBoneTreeNode(Bone bone) {
 			if (bone == null) return;
 
 			var children = bone.GetChildren();
+			var decendents = bone.GetDescendants();
+
+			bool hasChildInQuery = decendents.Any(BoneDoesMatch);
+			bool hasChildSelected = decendents.Any(Skeleton.IsBoneSelected);
+			bool isSelected = Skeleton.IsBoneSelected(bone);
+			bool isQueried = BoneDoesMatch(bone);
 
-			var flag = children.Count > 0 ? ImGuiTreeNodeFlags.OpenOnArrow : ImGuiTreeNodeFlags.Leaf;
-			if (Skeleton.IsBoneSelected(bone))
-				flag |= ImGuiTreeNodeFlags.Selected;
+			var criteria = HighlightCriteria.None;
 
-			var show = DrawBoneNode(bone, flag, () => OverlayWindow.SetGizmoOwner(bone.UniqueName));
-			if (show) {
+			if (isSelected) criteria ^= HighlightCriteria.Selected;
+			if (isQueried) criteria ^= HighlightCriteria.Queried;
+			if (hasChildSelected) criteria ^= HighlightCriteria.ChildSelected;
+			if (hasChildInQuery) criteria ^= HighlightCriteria.ChildQueried;
```

Comment:
Use OR instead of XOR here (same behavior, but with OR the intent of setting flags is clearer)

```suggestion
			if (isSelected) criteria |= HighlightCriteria.Selected;
			if (isQueried) criteria |= HighlightCriteria.Queried;
			if (hasChildSelected) criteria  |= HighlightCriteria.ChildSelected;
			if (hasChildInQuery) criteria |= HighlightCriteria.ChildQueried;
```